compiler/rustc_mir_transform/src/coroutine.rs

```diff
@@ -1057,7 +1057,7 @@ fn insert_switch<'tcx>(
     let blocks = body.basic_blocks_mut().iter_mut();
 
     for target in blocks.flat_map(|b| b.terminator_mut().successors_mut()) {
```

Comment:
can this be changed into a `count`?

Reply:
This loops over a bunch of `BasicBlock`s and increments each one, so something like `*target += blocks.flat_map(..).count()` would mean something completely different.
Unless I misunderstood you?

Reply:
Oh, I'm blind, lol. I thought we were incrementing a counter, not incrementing a per-bb index.